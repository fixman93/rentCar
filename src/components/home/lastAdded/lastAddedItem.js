import React, { Component } from 'react'
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom'

export class LastAddedItem extends Component {

  render() {
    const { project } = this.props
    console.log('eeeeee', project)
    return (
      <section className='last-added-item'>
        {project && project.id ?
          <Link to={'/project/' + project.id} >
            <img src={project.userImage} alt='Car' />
          </Link> : null}
        {project && project.carType ? <h3>{project.carType} {project.carModel} <span>{project.carYear}</span></h3> : null}
        <div className='vihicle-price'>
          {project && project.Currency ?
            <div className='currency'>{project.Currency}</div> :
            null}
          {project && project.carPrice ?
            <div className='price'>{project.carPrice}</div> :
            null}
          <span>per day</span>
        </div>
      </section>
    )
  }
}

LastAddedItem.propTypes = {
  project: PropTypes.object.isRequired
};


export default LastAddedItem 