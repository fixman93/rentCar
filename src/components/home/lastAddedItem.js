import React, { Component } from 'react'

class LastAddedItem extends Component {

  render() {

    return (
      <section className='last-added-item'>
        car
      </section>
    )
  }
}


export default LastAddedItem