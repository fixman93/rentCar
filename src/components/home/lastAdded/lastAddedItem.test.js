import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'

import { LastAddedItem } from './lastAddedItem'

configure({ adapter: new Adapter() });

describe('lastAddedItem', () => {
  const props = { project: { Currency: 20, id: 1 } }
  const lastAddedItem = shallow(<LastAddedItem {...props} />)
  // console.log('aaaa', lastAddedItem.instance().props)
  console.log('aaaaaa', lastAddedItem.debug())
  it('render lastAddedItem', () => {
    expect(lastAddedItem).toMatchSnapshot()
  })

  it('displays the currency from props', () => {
    expect(lastAddedItem.find('.currency').text()).toEqual('20')
  })

  it('contains a section', () => {
    expect(lastAddedItem.find('.last-added-item').exists()).toBe(true)
  })

})