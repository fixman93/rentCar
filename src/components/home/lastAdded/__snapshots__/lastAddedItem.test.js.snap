// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lastAddedItem render lastAddedItem 1`] = `ShallowWrapper {}`;
